controller/codebase.go

```diff
@@ -107,15 +113,22 @@ func (c *CodebaseController) Edit(ctx *app.EditCodebaseContext) error {
 	var existingWorkspaces []*app.Workspace
 	for _, workspace := range workspaces {
 		openLink := rest.AbsoluteURL(ctx.Request, fmt.Sprintf(app.CodebaseHref(cb.ID)+"/open/%v", workspace.Config.Name))
+
+		ideLink := workspace.GetHrefByRelOfWorkspaceLink(che.IdeUrlRel)
+		selfLink := workspace.GetHrefByRelOfWorkspaceLink(che.SelfLinkRel)
+
 		existingWorkspaces = append(existingWorkspaces, &app.Workspace{
 			Attributes: &app.WorkspaceAttributes{
+				ID:          &workspace.ID,
 				Name:        &workspace.Config.Name,
-				Description: &workspace.Status,
+				Description: &workspace.Description,
 				Status:      &workspace.Status,
 			},
 			Type: "workspaces",
 			Links: &app.WorkspaceLinks{
 				Open: &openLink,
+				Self: &selfLink,
+				Ide:  &ideLink,
```

Comment:
Why bypass the 'open' flow here by providing a link to the remote server Workspace directly?

Reply:
@aslakknutsen take a look at https://github.com/openshiftio/openshift.io/issues/1439#issuecomment-364901425
we want to communicate with with che-server directly from UI for most of the operations in future.

Reply:
@ibuziuk for the record: I fundamentally disagree.